pkg/processor/trigger/v3iostream/trigger.go

```diff
@@ -222,6 +225,15 @@ func (vs *v3iostream) ConsumeClaim(session streamconsumergroup.Session, claim st
 	return submitError
 }
 
+func (vs *v3iostream) Abort(session streamconsumergroup.Session) error {
+	vs.Logger.Debug("Abort called in v3iostream trigger")
```

Comment:
move to `triggers/trigger.go` (Abstract Trigger). on abstract function you receive no args. and it `vs.restartChan <- vs` only (before aborting, log with warning). then, from here, create a private function `abort` that will simply call the abstract class with `.Abort()`.

Reply:
This is a bit tricky since `AbstractTrigger` does not implement `Trigger` type.
Also, `Abort` is a part of the v3io `Handler` interface, which only v3iostream trigger implements.
So - is it really necessary to move it to Abstract Trigger? WDYT?

Reply:
`Trigger` and `AbstractTrigger` should have some "Restart" interface functionality. v3iotrigger itself will call that `.Restart()` upon `.abort()` execution. The reason I suggest such flow is to allow other triggers to be able to restart themselves more easily. 
Remember, the idea is to consolidate flows, not to diverge. so in the future, when we will need to restart another trigger, it will already be there for us.